### ISSUE TITLE
Feature request: completely redo TLC and SANY command line interfaces

### ISSUE THREAD

**Issue by ahelwer** [tlaplus/tlaplus#783]
This would be a good project for a new contributor. Leave the legacy command line interfaces around if people want to keep using them, but provide alternative front-ends with the following:

1. Parsed with a standard command line argument parsing library, following conventions people expect; for example:
   * short `-h` and long `--help` versions of commands
   * automatically print out usage and help if commands are incorrect
   * use subcommands instead of mutually-incompatible flags and options
2. Use reasonable defaults; for example, use the equivalent of the following by default:
   * `-workers auto`
   * `-lncheck final`
3. Return well-known error codes for easier integration with CI and scripts (TLC seems to mostly do this but SANY needs work; see #781)
4. Expose a simplified set of parameters; power users can continue to use the legacy command line if they want

**Comment by lemmy** [comment]
The current command-line is not great from a UX perspective and a more principled approach would make TLC more accessible to new users.  However, I propose for this issue to be more ambitious.  The major advantage of wrappers such as tlacli and tlabin is that they allow users to (mostly) work around TLC's config files.   Although we don't want users to write TLA+ on the command-line, it would be a major improvement if we find a more flexible replacement for TLC's config file that is exposed via the command-line.

**Comment by ahelwer** [comment]
I actually like TLC config files, they are quite simple. It is just difficult to know the syntax due to lack of learning material; everything uses the toolbox to generate them. I often find myself referring to the actual cfg file grammar in specifying systems when I write them myself.

**Comment by lemmy** [comment]
Indeed, config files are simple... actually they are too simple, forcing us to write an MC.cfg *and* an MC.tla file for anything serious.

**Comment by lemmy** [comment]
Related: https://github.com/tlaplus/tlaplus/issues/781

**Comment by ahelwer** [comment]
Wanted to highlight one example of how this issue affects me day-to-day: I cannot for the life of me remember all these names

- `tlc2.TLC`
- `tla2sany.SANY`
- `pcal.trans`

I have written them down in my notebook and have to look them up every time I want to run the tools. They are not easy to remember nor discover.

**Comment by ahelwer** [comment]
@lemmy what do you think about changing the `java -jar tla2tools.jar` behavior to invoke this more advanced command line interface instead of just wrapping `tlc2.TLC`?

**Comment by lemmy** [comment]
By now, the current behavior (default to tlc2.TLC) is used all over the place.  I'd also assume that most users only ever invoke TLC, and that SANY, PCal, ... typically are called from IDEs.  That said, TLC.java could redirect a command line that starts with `tla.SANY` or `tla.PCAL`.

**Comment by Calvin-L** [comment]
@ahelwer You could add a fifth point to the issue description: a new CLI would ideally also set various performance-related defaults like `-Dtlc2.tool.fp.FPSet.impl=tlc2.tool.fp.OffHeapDiskFPSet` (see discussion in #833). Ideally it would also automatically set recommended JVM flags like `-XX:+UseParallelGC`. That flag is tricky because it cannot be set from within the JVM itself.

**Comment by goyalyashpal** [comment]
> I cannot for the life of me remember all these names
> 
> * `tlc2.TLC`
> * `tla2sany.SANY`
> * `pcal.trans`
> 
> I have written them down in my notebook and have to look them up every time I want to run the tools
>
> \- _@ ahelwer at https://github.com/tlaplus/tlaplus/issues/783#issuecomment-1437118334_

if you are on windows 7+ os, then the scoop package i've created for tla-toolbox-pre makes it very simple to run them with names like follows.  scoop allows to specify arbitrary names, so these could have been even simpler (suggestions welcome) - but i picked these to avoid diverging from upstream:

* `tlc`
* `tla2sany`
* `pcal-trans`
* `tla2tex`
* `tlc-repl`

Check it out here:
* https://github.com/ScoopInstaller/Versions/pull/2669/changes#diff-2a9fb89d895d1dfbbbc803fbe40e2919f04289f9588ccddcbe3f4c1f15aa6758R40-R64
